packages/babel/test/as-input-plugin.js

```diff
@@ -0,0 +1,430 @@
+  const bundle = await rollup({
+    input: 'fixtures/class/main.js',
+    plugins: [babelPlugin({ babelHelpers: 'bundled' })]
+  });
+
+  const {
+    output: [{ code, map }]
+  } = await bundle.generate({ format: 'cjs', sourcemap: true });
+
+  const target = 'log';
+  const smc = new SourceMapConsumer(map);
+  const loc = getLocation(code, code.indexOf(target));
+  const original = smc.originalPositionFor(loc);
+
+  t.deepEqual(original, {
+    source: 'fixtures/class/main.js'.split(nodePath.sep).join('/'),
+    line: 3,
+    column: 12,
+    name: target
+  });
+});
+
+test('works with proposal-decorators (#18)', async (t) => {
+  await t.notThrowsAsync(() =>
+    rollup({
+      input: 'fixtures/proposal-decorators/main.js',
+      plugins: [babelPlugin({ babelHelpers: 'bundled' })]
+    })
+  );
+});
+
+test('checks config per-file', async (t) => {
+  const code = await generate('fixtures/checks/main.js', {}, { format: 'esm' });
+  t.true(code.includes('class Foo'));
+  t.true(code.includes('var Bar'));
+  t.false(code.includes('class Bar'));
+});
+
+test('allows transform-runtime to be used instead of bundled helpers', async (t) => {
+  const warnings = [];
+  const code = await generate(
+    'fixtures/runtime-helpers/main.js',
+    { babelHelpers: 'runtime' },
+    {},
+    {
+      onwarn(warning) {
+        warnings.push(warning.message);
+      }
+    }
+  );
+  t.deepEqual(warnings, [
+    `'@babel/runtime/helpers/classCallCheck' is imported by fixtures${nodePath.sep}runtime-helpers${nodePath.sep}main.js, but could not be resolved – treating it as an external dependency`
+  ]);
+  t.is(
+    code,
+    `'use strict';
+
+function _interopDefault (ex) { return (ex && (typeof ex === 'object') && 'default' in ex) ? ex['default'] : ex; }
+
+var _classCallCheck = _interopDefault(require('@babel/runtime/helpers/classCallCheck'));
+
+var Foo = function Foo() {
+  _classCallCheck(this, Foo);
+};
+
+module.exports = Foo;
+`
+  );
+});
+
+test('allows transform-runtime to inject esm version of helpers', async (t) => {
+  const warnings = [];
+  const code = await generate(
+    'fixtures/runtime-helpers-esm/main.js',
+    { babelHelpers: 'runtime' },
+    {
+      format: 'esm'
+    },
+    {
+      onwarn(warning) {
+        warnings.push(warning.message);
+      }
+    }
+  );
+  t.deepEqual(warnings, [
+    `'@babel/runtime/helpers/esm/classCallCheck' is imported by fixtures${nodePath.sep}runtime-helpers-esm${nodePath.sep}main.js, but could not be resolved – treating it as an external dependency`
+  ]);
+  t.is(
+    code,
+    `import _classCallCheck from '@babel/runtime/helpers/esm/classCallCheck';
+
+var Foo = function Foo() {
+  _classCallCheck(this, Foo);
+};
+
+export default Foo;
+`
+  );
+});
+
+test('allows transform-runtime to be used instead of bundled helpers, but throws when CommonJS is used', async (t) => {
+  await t.throwsAsync(
+    () => generate('fixtures/runtime-helpers-commonjs/main.js', { babelHelpers: 'runtime' }),
```

Comment:
I think `throwsAsync` takes a Promise instead of a function. `throws` takes a function.

Reply:
This test wouldnt pass if expected parameters would be different, but you are right - it can receive a promise, but can also receive an async function (which this kinda is):
https://github.com/avajs/ava/blob/master/docs/03-assertions.md#throwsasyncthrower-expected-message

I can change this if you'd like, but personally I kinda like passing functions to every `throws*`-related assertion - less thinking about what "expected" parameter is.

Reply:
I wasn't aware that you could also pass an async function. In that case you can leave it as it currently is :)